test/test_ff_utils.py

```diff
@@ -227,12 +235,114 @@ def get_access_keys(self):  # noqa - this is a mock so PyCharm shouldn't suggest
             assert 'Must provide a valid authorization key or ff' in str(exec_info.value)
 
 
-# Integration tests
+SSE_CUSTOMER_KEY = 'SSECustomerKey'
+SSE_CUSTOMER_ALGORITHM = 'SSECustomerAlgorithm'
 
+foo_env = 'fourfront-foo'
+foo_env_auth_key, foo_env_auth_secret = foo_env_auth_tuple = ('fookey', 'foosecret')
+foo_env_auth_dict = {
+    'key': foo_env_auth_key,
+    'secret': foo_env_auth_secret
+}
 
-@pytest.mark.integrated
+bar_env = 'fourfront-bar'
+bar_env_auth_key, bar_env_auth_secret = bar_env_auth_tuple = ('barkey', 'barsecret')
+bar_env_url = 'http://fourfront-bar.example/'
+bar_env_url_trimmed = bar_env_url.rstrip('/')
+bar_env_auth_dict = {
+    'key': bar_env_auth_key,
+    'secret': bar_env_auth_secret,
+    'server': bar_env_url,
+}
+bar_env_default_auth_dict = {'default': bar_env_auth_dict}
+
+some_server = 'http://localhost:8000'
+some_auth_key, some_auth_secret = some_auth_tuple = ('mykey', 'mysecret')
+some_badly_formed_auth_tuple = ('mykey', 'mysecret', 'other-junk')  # contains an extra element
+some_auth_dict = {'key': some_auth_key, 'secret': some_auth_secret}
+some_auth_dict_with_server = {'key': some_auth_key, 'secret': some_auth_secret, 'server': some_server}
+some_badly_formed_auth_dict = {'kee': some_auth_key, 'secret': some_auth_secret}  # 'key' is misspelled
```

Comment:
Seems like these things belong in a fixture or something... They feel out of place here.

Reply:
It's almost an anti-pattern here because putting them in a fixture would mean I'd have to unpack them from the fixture. The whole point is just to have these values set up and ready to use by several tests. However, I did some light rearranging and added some comments explaining, so maybe it won't be so jarring.